app/controllers/pages/questions_controller.rb

```diff
@@ -1,25 +1,20 @@
 class Pages::QuestionsController < PagesController
   def new
-    @question_form = Pages::QuestionForm.new(form_id: current_form.id,
-                                             answer_type: draft_question.answer_type,
+    @question_form = Pages::QuestionForm.new(answer_type: draft_question.answer_type,
                                              question_text: draft_question.question_text,
                                              answer_settings: draft_question.answer_settings,
                                              is_optional: draft_question.is_optional,
                                              draft_question:)
 
+    # TODO: Remove this once we have a check your question view. The new view should also pull data directly from draft_question instead of through page model
     @page = Page.new(form_id: current_form.id)
     render :new, locals: { current_form:, draft_question: }
   end
 
   def create
-    @question_form = Pages::QuestionForm.new(page_params.merge(answer_settings: draft_question.answer_settings,
-                                                               page_heading: draft_question.page_heading,
-                                                               guidance_markdown: draft_question.guidance_markdown,
-                                                               draft_question:))
-    @page = Page.new(page_params.merge(answer_settings: draft_question.answer_settings,
-                                       page_heading: draft_question.page_heading,
-                                       guidance_markdown: draft_question.guidance_markdown,
-                                       answer_type: draft_question.answer_type))
+    @question_form = Pages::QuestionForm.new(page_params.merge(page_params_for_form_object))
```

Comment:
I don't think we need to do `page_params.merge` here? It looks like `page_params_for_form_object` already does it.

```suggestion
    @question_form = Pages::QuestionForm.new(page_params_for_form_object)
```

Reply:
Good catch, thanks I'll push up another commit to fix this

Reply:
I decided to squash the commit to save you having to approve it again once the two commits were squashed.